### DISCUSSION TITLE
How to fix Joomla flagging custom component/module as potentially incompatible

### DISCUSSION THREAD

**Discussion by Chaosxmk** [joomla/joomla-cms#36743] [Q&A] [answered]
Whenever Joomla 4 has an update that checks compatibility, my component and modules gets flagged as potentially incompatible. Before every Joomla update, I test all of my custom stuff on a dev site, so I know that everything does work in Joomla 4, but I don't see what exactly it is that is flagging them, is there something somewhere I can change to fix this?

**Comment by richard67** [comment, marked as answer]
Your extension has to support the Joomla update system and so have an update site, and the extension XML has to contain the right `<targetplatform>` tag, that's all.

Details about tha XML file and the targetplaform tag you can find here: https://docs.joomla.org/Deploying_an_Update_Server

The pre-update checker only check that targetplatform tag of the extension (details) XML, nothing else. It does not do any checks of local code and does not check any local manifest XML files.

I hope my answer helps. If it answrs your question, please mark my answer so people can see the question was answered. Thanks in advance.
